countries_crawler/src/asyncio_gather.py

```diff
@@ -0,0 +1,58 @@
+"""Web Scrappper with Asyncio"""
+
+import json
+import asyncio
+import aiohttp
+from utils import parser_country, parse_countries_urls
+
+
+async def get_response(session, url):
+    """Get Response from a URL
+
+    This method receive a session and url of wikipedia and return a response.
+    This will also handle the connection error and HTTPError.
+    """
+
+    async with session.get(url, ssl=False) as response:
+        return {
+            "url": url,
+            "status_code": response.status,
+            "response_text": await response.text(),
+        }
+
+
+async def display_country_detail(session, url):
+    """Coroutine method to get data from url
+
+    This method is a coroutine. This should be called by asyncio. This function
+    receive a session and url. First, it calls get_response() and wait for
+    response. When a response is returned by get_response() it sends this
+    response to parser to extract the required information from the html page.
+    And finally print the result of parser() in JSON format.
+    """
+    response = await get_response(session, url)
+    country = parser_country(
+        response["response_text"], response["url"], response["status_code"]
```

Comment:
can't we just pass the whole response object?

Reply:
We can pass whole `response` and `url` to `parser_country()`.
Actually, we have two types of response `requests response` and `aiohttp response`. And we need only HTML content to parse the country details. Both responses have different functions to get HTML contents like:
- In `aiohttp` response.text(), response.status
- in `requests` response.content, response.status_code
so we will have to make a check in `parser_country()`. It is better to make `parser_country()' only receive HTML contents and parse them.
Still, if you think `parser_country()` should receive the whole response then I will do it.